components/emotionalRegulation/TimeModal.tsx

```diff
@@ -0,0 +1,222 @@
+    content: {
+      title: 'KopfSACHEN',
+      body: 'Zeit deinen Tagesplan zu kontrollieren',
+    },
+    trigger: date,
+  });
+};
+
+/** need to ask for permission on ios */
+async function registerForPushNotificationsAsync() {
+  const { status: existingStatus } = await Notifications.getPermissionsAsync();
+  let finalStatus = existingStatus;
+  if (existingStatus !== 'granted') {
+    const { status } = await Notifications.requestPermissionsAsync();
+    finalStatus = status;
+  }
+  if (finalStatus !== 'granted') {
+    alert('Failed to get push token for push notification!');
+  }
+}
+
+export default function TimeModal({ toggle }: { toggle: () => void }) {
+  const [date, setDate] = useState(new Date());
+  const [mode, setMode] = useState('date');
+
+  // for iOS 14 bc there are some UI problems....
+  const displayOptions =
+    Platform.OS === 'ios' ? (Platform.Version.startsWith('14') ? 'default' : 'spinner') : 'default';
+
+  let style = {};
+  if (Platform.OS === 'ios') {
+    if (Platform.Version.startsWith('14')) {
+      style = {
+        flex: 0,
+        height: '100%',
+        width: '60%',
+        marginLeft: '30%',
+        alignSelf: 'center',
+        justifyContent: 'center',
+      };
+    }
+  }
+
+  const onChange = (event: DateTimePickerEvent, newDate?: Date | undefined) => {
+    const currentDate = newDate;
+    if (currentDate) {
+      setDate(currentDate);
+    }
+  };
+
+  const showMode = (currentMode: string) => {
+    setMode(currentMode);
+  };
+
+  const showDatepicker = () => {
+    showMode('date');
+  };
+
+  const showTimepicker = () => {
+    showMode('time');
+  };
+
+  /** alert poping up to ask again if user certain they want to receive
+   * a notification, to make it less frustrating
+   */
+  const handleSetNotification = () => {
+    Alert.alert('Benachrichtigung', `Du wirst um ${date.toLocaleString()} erinnert`, [
+      {
+        text: 'Cancel',
+        style: 'cancel',
+      },
+      {
+        text: 'OK',
+        onPress: () => {
+          triggerNotifications(date);
+          registerForPushNotificationsAsync();
+        },
+      },
+    ]);
+  };
+
+  return (
+    <Modal animationType="slide" transparent={true}>
+      <View style={styles.overlay}>
+        <View style={styles.popup}>
+          <View>
+            <Pressable
+              accessibilityHint="Datum auswählen"
+              onPress={showDatepicker}
+              style={[{ backgroundColor: mode === 'date' ? PRIMARY : WHITE }, styles.btn]}>
+              <Text style={styles.btnTxt}>Tag auswählen </Text>
+            </Pressable>
+          </View>
+          <View>
+            <Pressable
+              accessibilityHint="Tageszeit auswählen "
+              onPress={showTimepicker}
+              style={[{ backgroundColor: mode === 'time' ? PRIMARY : WHITE }, styles.btn]}>
+              <Text style={styles.btnTxt}>Uhrzeit auswählen </Text>
+            </Pressable>
```

Comment:
Hier ist die Setzung einer Benachrichtigung möglich welche in der Vergangenheit liegt. Schätze das sollte nicht erlaubt sein oder?

Reply:
Du hättest eine "Benachrichtigung"/Alert erhalten müssen, die dich fragt, ob expo dir notifications senden darf. Macht mir eher Sorgen, dass der Alert bei dir nicht erschienen ist. Bei mir geht das noch für iOS und Android...

Reply:
Ok, ich glaube ich konnte den Fehler beheben. Falls Du es schon lokal versuchen willst vor dem return:   
handleSetNotification damit ersetzten: 
const handleOnPressOk = () => {
    triggerNotifications(date);
  }
  /** alert poping up to ask again if user certain they want to receive
   * a notification, to make it less frustrating
   */
  const handleSetNotification = () => {
    Alert.alert('Benachrichtigung', `Du wirst um ${date.toLocaleString()} erinnert`, [
      {
        text: 'Cancel',
        style: 'cancel',
      },
      {
        text: 'OK',
        onPress: handleOnPressOk,
      },
    ]);
  };
useEffect(() => {
    registerForPushNotificationsAsync();
  }, [])

Reply:
Yes sieht jetzt gut aus, habe eine Benachrichtigung bekommen.